diffpy/snmf/factorizers.py

```diff
@@ -0,0 +1,29 @@
+import numpy as np
+import scipy.optimize
+
+
+def lsqnonneg(coefficient, target):
+    """
+    Solve ``argmin_x || Ax - b ||_2`` for ``x>=0``.
```

Comment:
this statement is very clear, but below i have a question about whether any of the arrays passed to the function need to be  positive.  What behavior do you want  if a matrix containing negative numbers is passed in?

Reply:
I will take another look at the original code and add a test case for this.